cdap-ui/app/features/hydratorplusplus/controllers/create/leftpanel-ctrl.js

```diff
@@ -83,14 +90,13 @@ class HydratorPlusPlusLeftPanelCtrl {
           };
           this.leftpanelStore.dispatch(this.leftpanelActions.fetchPlugins(ext, params));
         } else {
-          this.pluginsMap
-              .filter( pluginObj => pluginObj.name === this.HydratorPlusPlusOrderingFactory.getPluginTypeDisplayName(ext))
-              .forEach( matchedObj => {
-                let getPluginTemplateNode = (ext) => {
-                  return this.leftpanelStore.getState().plugins.pluginTypes[ext];
-                };
-                matchedObj.plugins = getPluginTemplateNode(ext);
-              });
+          fetchPluginsFromMap(ext)
+            .forEach( matchedObj => {
+              matchedObj.plugins = matchedObj
+                .pluginTypes
+                .map(pluginType => this.leftpanelStore.getState().plugins.pluginTypes[pluginType] || [])
+                .reduce((prev, curr) => prev.concat(curr), []);
+            });
```

Comment:
I feel like most of the content of the subscriber should be moved into the left panel store. This function is getting called about 10 times when you open hydrator, and it seems like it is wasteful to keep reordering, and mapping for every change from left-panel-store. We should be able to make the calls from the extensions, and concat the result to a pluginsMap within the store. The subscriber then can be light weight, and simply fetch the pluginsMap, and run the orderingPluginTypes function.  At the moment this subscriber is quite confusing